Tribler/community/tunnel/tunnel_community.py

```diff
@@ -277,24 +290,47 @@ def start_download_test(self):
-        # pub-sha1 28bad8f1722de818700228c471ab0d7786736c05
-        # -----BEGIN PUBLIC KEY-----
-        # MIGnMBAGByqGSM49AgEGBSuBBAAnA4GSAAQDq2xd3qRPgG4/xYHqkMw4DXNdLCTK
-        # h7OmjN8fld5QAQrv8xI0Tu3K4buWA42xYL5LMaICiQIyUaKtiLHxi94PFASYQ/Oj
-        # XfUBdjCkReY7tQBWIcBMm3uoyozOvlZ7NVtrIdhGGZvY3MnnBI3Fm/Pxq3A3Lxno
-        # XTwhM/VlefpRCIQPUv9Oo+QZBmI8HW6OLq8=
-        # -----END PUBLIC KEY-----
-        master_key = "3081a7301006072a8648ce3d020106052b81040027038192000403ab6c5ddea44f806e3fc581ea90cc380d735d2c24ca87b3a68cdf1f95de50010aeff312344eedcae1bb96038db160be4b31a20289023251a2ad88b1f18bde0f14049843f3a35df5017630a445e63bb5005621c04c9b7ba8ca8ccebe567b355b6b21d846199bd8dcc9e7048dc59bf3f1ab70372f19e85d3c2133f56579fa5108840f52ff4ea3e41906623c1d6e8e2eaf".decode(
+        #generated: Mon Mar  9 16:21:28 2015
+        #curve: None
+        #len: 571 bits ~ 144 bytes signature
+        #pub: 170 3081a7301006072a8648ce3d020106052b81040027038192000404dc19d38890e0de983aed312d0b0c8a27732d7499a3e0c5d7bebfb8451270215d788ca671040935b4b4fc7faa48fd021226f5580995d63d0e9c82b0586850f93768debf550f4459054e6fb91318d8a0346c4059a4e84c95e4b7769cadc296d567ad353752a630d20f077a9f068998136338f2f0663d327d8934110565fb41040ac2d94c4fb78308118206a3930b68a8
+        #pub-sha1 3df2df7afa551c6d876a94b44c4b37d417b7c4e8
+        #-----BEGIN PUBLIC KEY-----
+        #MIGnMBAGByqGSM49AgEGBSuBBAAnA4GSAAQE3BnTiJDg3pg67TEtCwyKJ3MtdJmj
+        #4MXXvr+4RRJwIV14jKZxBAk1tLT8f6pI/QISJvVYCZXWPQ6cgrBYaFD5N2jev1UP
+        #RFkFTm+5ExjYoDRsQFmk6EyV5Ld2nK3CltVnrTU3UqYw0g8Hep8GiZgTYzjy8GY9
+        #Mn2JNBEFZftBBArC2UxPt4MIEYIGo5MLaKg=
+        #-----END PUBLIC KEY-----
+        master_key = "3081a7301006072a8648ce3d020106052b81040027038192000404dc19d38890e0de983aed312d0b0c8a27732d7499a3e0c5d7bebfb8451270215d788ca671040935b4b4fc7faa48fd021226f5580995d63d0e9c82b0586850f93768debf550f4459054e6fb91318d8a0346c4059a4e84c95e4b7769cadc296d567ad353752a630d20f077a9f068998136338f2f0663d327d8934110565fb41040ac2d94c4fb78308118206a3930b68a8".decode(
             "HEX")
         master = dispersy.get_member(public_key=master_key)
         return [master]
 
     def initiate_meta_messages(self):
-        return super(TunnelCommunity, self).initiate_meta_messages() + \
+        messages = super(TunnelCommunity, self).initiate_meta_messages()
+        copy_messages = messages[:]
+        for message in copy_messages:
+            if message.name == "dispersy-introduction-request" or message.name == "dispersy-introduction-response":
+                messages.remove(message)
+            
+        messages += [Message(self, u"dispersy-introduction-request",
```

Comment:
Have a look at how we're replacing the the payload in the discovery community
https://github.com/Tribler/dispersy/blob/devel/discovery/community.py#L232
I think its slightly nicer